test/Client.Tests/ConnectionTests.cs

```diff
@@ -85,5 +85,165 @@ public ConnectionTests(ITestOutputHelper testOutput)
 
             await Task.WhenAll(clientConnection.HasHasDisconnected, serverConnection.HasHasDisconnected);
         }
+
+        /// <summary>
+        ///     Verify that a client <see cref="LspConnection"/> can handle a request from a server <see cref="LspConnection"/>.
+        /// </summary>
+        [Fact(DisplayName = "Client connection can handle request from server")]
+        public async Task Server_HandleRequest_Success()
+        {
+            LspConnection clientConnection = await CreateClientConnection();
```

Comment:
Minor nitpick, if we make `CreateClientConnection` return a disposable we can wrap all these calls in a using and remove some boilerplate with disconnect.

However that can be changed later.